week6/Filmer/app/src/main/java/com/tahhan/filmer/repository/MovieRepo.kt

```diff
@@ -22,30 +23,21 @@ class MovieRepo(application: Application) {
     }
 
     var movieList: MutableLiveData<List<Movie>>? = null
-    var movie : MutableLiveData<Movie>? = MutableLiveData()
-
+    var movie: MutableLiveData<Movie>? = MutableLiveData()
     private var database: MovieDB = MovieDB.getDatabase(application)
 
-    fun getFavoriteMovieList(): LiveData<List<Movie>> {
-        return database.movieDao().loadFavoriteMovies()
-    }
+    fun getFavoriteMovieList(): Flow<List<Movie>> = database.movieDao().loadFavoriteMovies()
 
-    fun getFavoriteMovieByID(id: String): MutableLiveData<Movie>? {
-        GlobalScope.launch {
-            movie?.postValue(database.movieDao().loadMovieById(id))
-        }
-        return movie
+    fun getFavoriteMovieByID(id: String): Flow<Movie> = database.movieDao().loadMovieById(id)
 
-    }
 
-    fun insertMovie(movie: Movie) {
-            database.movieDao().insertMovie(movie)
+    suspend fun insertMovie(movie: Movie) {
+        database.movieDao().insertMovie(movie)
```

Comment:
If you had the DAO, you could've just called functions directly, instead of accessing the DAO every time.